docs/howto/system_testing.md

```diff
@@ -434,6 +434,10 @@ for system tests.
 | skip_transform_validation | boolean |  | Disable or enable the transforms validation performed in system tests. |
 | vars | dictionary |  | Package level variables to set (i.e. declared in `$package_root/manifest.yml`). If not specified the defaults from the manifest are used. |
 | wait_for_data_timeout | duration |  | Amount of time to wait for data to be present in Elasticsearch. Defaults to 10m. |
+| assert.hit_count | integer |  | Exact number of documents to wait for being ingested. |
+| assert.min_count | integer |  | Minimum number of documents to wait for being ingested. |
```

Comment:
Currently, the maximum number of documents retrieved from Elasticsearch by elastic-package is 500:
https://github.com/elastic/elastic-package/blob/b6f296ce42c19fa4c6a58da623e5436dc727d1b5/internal/testrunner/runners/system/tester.go#L85

I guess it's difficult to happen, but it's worth to take this into account. If a test sets `assert.min_count: 510` , that will never finish since there will never be 510 documents in the search response.

Related: https://github.com/elastic/elastic-package/issues/2378#issuecomment-2674940199

As mentioned in that comment, should we add a new environment variable to be able to update that value (in a follow-up PR maybe)?

Reply:
Umm, I guess it is also possible to use some kind of pagination? Maybe in the meantime we can also limit `hit_count` and `min_count` to 500 documents, and fail if it is set to a bigger number.

Reply:
Removed and updated PR description too.